mysql_mimic/results.py

```diff
@@ -200,10 +207,10 @@ def _binary_encode_timedelta(val):
 }
 
 
-def binary_encode(val):
```

Comment:
Should we just be using the given type to encode values? Instead of inferring?

Reply:
we could have a separate dict for binary encoders / text encoders / etc but this feels equivalent / more compact imo